src/app/mapView/MapView.jsx

```diff
@@ -1,72 +1,113 @@
-    radius: defaultRadius,
+    suggestions: []
   };
 
-  componentWillMount() {
-    this.fetchLocations();
+  onSearchChanged = event => {
+    const searchString = event.target.value;
+    if(searchString){ 
+      this.onSuggestionsFetchRequested({searchString});
+    } else {
+      this.onSuggestionsClearRequested();
+    }
+  };
+
+  onBoundsChanged = ({center, radius}) => {
+    if(center.lat() === this.state.center.lat && 
+        center.lng() === this.state.center.lng) return;
+    this.fetchLocations(center, radius);
   }
 
-  componentDidMount() {
-    if (!navigator || !navigator.geolocation) {
-      return;
-    }
+  onSuggestionsFetchRequested = ({ searchString, reason }) => {
+    getOrganizations(searchString)
+      .then(organizations => { 
+          this.setState({ suggestions : organizations })
+      })
+      .catch(e => console.error('error', e));
+  };
 
-    navigator.geolocation.getCurrentPosition(
-      userPosition => {
-        const { coords } = userPosition;
-        this.onCenterChanged({
-          lat: coords.latitude,
-          lng: coords.longitude,
+  handleSuggestionClick = (organization) => {
+    getOrganizationLocations(organization.id)  
+      .then(locations => { 
+        if(!locations.length) return;
+        locations = locations.map( loc => ({...loc, Organization : organization}))   //add orgranization to locations
+        const firstLoc = locations[0];
+        const coords = firstLoc.position.coordinates;     //TODO: focus all
+        this.setState({ 
+          locations, 
+          suggestions: [], 
+          center: {
+            lat: coords[1],
+            lng: coords[0]  
+          }
         });
-      },
-      e => console.error('Failed to get current position', e),
-      { timeout: geolocationTimeout },
-    );
+        this.map.onToggleMarkerInfo(firstLoc.id);
+      })
+      .catch(e => console.error('error', e));
   }
-  onSearchChanged = event => {
-    this.setState({ searchString: event.target.value }, () => {
-      this.fetchLocations();
-    });
-  };
 
-  onCenterChanged = center => {
-    this.setState({ center }, () => {
-      this.fetchLocations();
+  onSuggestionsClearRequested = () => {
+    this.setState({
+      suggestions: []
     });
   };
 
-  fetchLocations = debounce(() => {
+  fetchLocations = (center, radius) => {
     getLocations({
-      latitude: this.state.center.lat,
-      longitude: this.state.center.lng,
-      radius: this.state.radius,
-      searchString: this.state.searchString,
+      latitude: center.lat(),
+      longitude: center.lng(),
+      radius: Math.floor(radius)
     })
-      .then(locations => this.setState({ locations }))
+      .then(locations => this.setState({ locations }))    //TODO: we can save these in the redux store
       .catch(e => console.error('error', e));
-  }, fetchLocationsDebouncePeriod);
+  };
 
   render() {
     return (
       <div className="Map">
+        <ul className="suggestions"
```

Comment:
Since we'll need this dropdown UI in other screens as well, it should probably be its own component.

Reply:
Agreed.

Reply:
I am going to punt on this until the next PR I submit, which will include the work to better support saving Organization data in the location flow. I'll do the UI work to support autocomplete of the organization name.